internal/app/middleware/middleware.go

```diff
@@ -0,0 +1,25 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+	"go.uber.org/zap"
+
+	logging "github.com/Te8va/APIbook/internal/pkg/logger"
+)
+
+func Log(next httprouter.Handle) httprouter.Handle {
+	return func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+		logging.Logger().Info("Request", zap.String("URL", r.URL.Path), zap.String("Method", r.Method))
+
+		defer func() {
+			if err := recover(); err != nil {
```

Comment:
do you have ```panic()``` somewhere?

Reply:
found it
but if logger is not initialized, I think, we can just stop the application (but it depends on importance of logs)